community/modules/compute/schedmd-slurm-gcp-v6-nodeset/main.tf

```diff
@@ -175,3 +175,23 @@ data "google_compute_reservation" "reservation" {
     # Add a validation that if reservation.project != var.project_id it should be a shared reservation
   }
 }
+
+data "google_compute_machine_types" "machine_types_by_zone" {
+  for_each = local.zones
+  filter   = format("name = \"%s\"", local.nodeset.machine_type)
+  zone     = each.value
+}
+
+locals {
+  machine_types_by_zone   = data.google_compute_machine_types.machine_types_by_zone
+  zones_with_machine_type = [for k, v in local.machine_types_by_zone : k if length(v.machine_types) > 0]
+}
+
+resource "terraform_data" "machine_type_zone_validation" {
+  lifecycle {
```

Comment:
You should be chaining this resource into the graph dependencies.

https://developer.hashicorp.com/terraform/language/resources/terraform-data

I suggest adding:

```hcl
  input = var.machine_type
```

and then replace line 80 with

```
    machine_type = terraform_data.machine_type_zone_validation.output
```